cmd/coi.go

```diff
@@ -0,0 +1,249 @@
+package cmd
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/exoscale/egoscale"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+const coiTemplateName = "Container-Optimized Instance"
+
+var coiCmd = &cobra.Command{
+	Use:   "coi <name>",
+	Short: "Deploy a Container-Optimized Instance",
+	Long: fmt.Sprintf(`This command creates a Compute instance running one or several Docker
+containers based on the Docker image specified or a Docker Compose
+configuration provided. It can be invoked in 2 ways:
+
+* Specifying a Docker image and optionally ports to publish:
+
+    $ exo lab coi nginx \
+        --zone ch-gva-2 \
+        --image nginxdemos/hello \
+        --port 80:80 \
+        ...
+
+* Specifying a Docker Compose file:
+
+    $ exo lab coi my-app \
+        --zone ch-gva-2 \
+        --docker-compose /path/to/docker-compose.yml \
+        # or --docker-compose https://my.app/docker-compose.yml
+        ...
+
+Once created, the Compute instance can be managed like any other standard
+instance using the "exo vm" commands.
+
+The Docker Compose configuration reference can be found at this address:
+https://docs.docker.com/compose/compose-file/
+
+Supported output template annotations: %s`,
+		strings.Join(outputterTemplateAnnotations(&vmShowOutput{}), ", ")),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			cmdExitOnUsageError(cmd, "invalid arguments")
+		}
+
+		cmdSetZoneFlagFromDefault(cmd)
+
+		return cmdCheckRequiredFlags(cmd, []string{"zone"})
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var cloudInitUserdata string
+
+		z, err := cmd.Flags().GetString("zone")
+		if err != nil {
+			return err
+		}
+
+		zone, err := getZoneByNameOrID(z)
+		if err != nil {
+			return err
+		}
+
+		diskSize, err := cmd.Flags().GetInt64("disk-size")
+		if err != nil {
+			return err
+		}
+
+		sshKey, err := cmd.Flags().GetString("ssh-key")
+		if err != nil {
+			return err
+		}
+		if sshKey == "" {
+			sshKey = gCurrentAccount.DefaultSSHKey
+		}
+
+		sg, err := cmd.Flags().GetStringSlice("security-group")
+		if err != nil {
+			return err
+		}
+
+		securityGroups, err := getSecurityGroups(sg)
+		if err != nil {
+			return err
+		}
+
+		privnet, err := cmd.Flags().GetStringSlice("private-network")
+		if err != nil {
+			return err
+		}
+
+		privnets, err := getPrivnetList(privnet, zone.ID)
+		if err != nil {
+			return err
+		}
+
+		so, err := cmd.Flags().GetString("service-offering")
+		if err != nil {
+			return err
+		}
+
+		serviceOffering, err := getServiceOfferingByNameOrID(so)
+		if err != nil {
+			return err
+		}
+
+		template, err := getTemplateByNameOrID(zone.ID, coiTemplateName, "community")
```

Comment:
IMHO, You should use `Fedora CoreOS`, it's natively container optimized 😉

Reply:
Well, just checked, docker-compose is not installed by default on it 😕 only Docker